framework/src/problems/FEProblemBase.C

```diff
@@ -4231,6 +4231,16 @@ FEProblemBase::computeUserObjectsInternal(const ExecFlagType & type,
       joinAndFinalize(query.clone().condition<AttribInterfaces>(Interfaces::DomainUserObject));
     }
 
+    // if any userobject may have written to variables we need to close the aux solution
+    for (const auto & uo : userobjs)
+      if (auto euo = dynamic_cast<const ElementUserObject *>(uo);
+          euo && euo->hasWritableCoupledVariables())
+      {
+        _aux->solution().close();
+        _aux->system().update();
+        break;
+      }
```

Comment:
why does this only cast to element user objects? Couldn't any of the other uo types that were joined and finalized above have written to variables?

Reply:
See this restriction https://github.com/idaholab/moose/blob/2461ac474fed95dd4256698438d74103d1fe9f64/framework/src/interfaces/Coupleable.C#L858

Reply:
Yeah, what Logan said. Which UO types do you think this would be useful for? Anything running on sides maybe? That could be added easily if needed.